automerge/src/value.rs

```diff
@@ -158,29 +158,47 @@ pub(crate) enum DataType {
     Undefined,
 }
 
-#[derive(Serialize, PartialEq, Debug, Clone)]
+#[derive(Serialize, Debug, Clone)]
 #[serde(untagged)]
 pub enum ScalarValue {
     Bytes(Vec<u8>),
     Str(SmolStr),
     Int(i64),
     Uint(u64),
     F64(f64),
-    Counter(i64),
+    Counter(i64, i64, usize),
```

Comment:
Given that this is three integer fields rather than just a wrapper now I think it might be worth naming the fields, otherwise anytime someone reads this they need to go and look at the places the counter is used to figure out which of the fields means what. Named fields would also make the various pattern matches on this enum throughout the codebase a little easier to read.

I'm also a little concerned that it's in our public API and wonder whether it would be worth having a separate counter struct with crate private fields which we put in the `ScalarValue::Counter`, that way users don't know or care about these details.

Reply:
Good call - I made a struct `Counter` to contain this data.